pkg/apiserver/resource_manager/service.go

```diff
@@ -0,0 +1,106 @@
+
+	"github.com/pingcap/tidb-dashboard/pkg/apiserver/user"
+	"github.com/pingcap/tidb-dashboard/pkg/apiserver/utils"
+	"github.com/pingcap/tidb-dashboard/pkg/tidb"
+	"github.com/pingcap/tidb-dashboard/util/featureflag"
+	"github.com/pingcap/tidb-dashboard/util/rest"
+)
+
+type ServiceParams struct {
+	fx.In
+	TiDBClient *tidb.Client
+}
+
+type Service struct {
+	FeatureResourceManager *featureflag.FeatureFlag
+
+	params ServiceParams
+}
+
+func newService(p ServiceParams, ff *featureflag.Registry) *Service {
+	return &Service{params: p, FeatureResourceManager: ff.Register("resource_manager", ">= 7.0.0")}
+}
+
+func registerRouter(r *gin.RouterGroup, auth *user.AuthService, s *Service) {
+	endpoint := r.Group("/resource-manager")
+	endpoint.Use(
+		auth.MWAuthRequired(),
+		s.FeatureResourceManager.VersionGuard(),
+		utils.MWConnectTiDB(s.params.TiDBClient),
+	)
+	{
+		endpoint.GET("/information", s.GetInformation)
+		endpoint.GET("/config", s.GetConfig)
+		endpoint.GET("/calibrate", s.GetCalibrate)
+	}
+}
+
+// @Summary Get Resource Control enable config
+// @Router /resource-manager/config [get]
+// @Security JwtAuth
+// @Success 200 {string} enable
+// @Failure 401 {object} rest.ErrorResponse
+// @Failure 500 {object} rest.ErrorResponse
+func (s *Service) GetConfig(c *gin.Context) {
+	db := utils.GetTiDBConnection(c)
+	var enable string
+	err := db.Raw("SELECT @@GLOBAL.tidb_enable_resource_control as tidb_enable_resource_control").Find(&enable).Error
+	if err != nil {
+		rest.Error(c, err)
+		return
+	}
+	c.JSON(http.StatusOK, enable)
+}
+
+type TableRowDef struct {
+	Name      string `json:"name" gorm:"column:NAME"`
+	RuPerSec  string `json:"ru_per_sec" gorm:"column:RU_PER_SEC"`
+	Priority  string `json:"priority" gorm:"column:PRIORITY"`
+	Burstable string `json:"burstable" gorm:"column:BURSTABLE"`
+}
+
+// @Summary Get Information of Resource Groups
+// @Router /resource-manager/information [get]
+// @Security JwtAuth
+// @Success 200 {object} TableRowDef
+// @Failure 401 {object} rest.ErrorResponse
+// @Failure 500 {object} rest.ErrorResponse
+func (s *Service) GetInformation(c *gin.Context) {
+	db := utils.GetTiDBConnection(c)
+	cfg := &TableRowDef{}
+	err := db.Table("INFORMATION_SCHEMA.RESOURCE_GROUPS").Find(cfg).Error
+	if err != nil {
+		rest.Error(c, err)
+		return
+	}
+	c.JSON(http.StatusOK, cfg)
+}
+
+// @Summary Get calibrate of Resource Groups by workload
+// @Router /resource-manager/calibrate [get]
+// @Param workload query string true "workload" default("tpcc")
+// @Security JwtAuth
+// @Success 200 {string} resourceNums
+// @Failure 401 {object} rest.ErrorResponse
+// @Failure 500 {object} rest.ErrorResponse
+func (s *Service) GetCalibrate(c *gin.Context) {
+	w := c.Param("workload")
+	db := utils.GetTiDBConnection(c)
+
+	var resourceNums string
+	err := db.Raw("calibrate resource workload (?)", w).Scan(resourceNums).Error
+	if err != nil {
+		rest.Error(c, err)
+		return
+	}
+	c.JSON(http.StatusOK, resourceNums)
```

Comment:
Maybe in the response, we can include the warnings, totalRU information as well, likes: 

```
{
    estimatedCapacity: xxx,
    totalRU: xxx,
    warnings: ['', '']
}
```

I'm not sure whether `warnings` and `totalRU` should hardcode in the frontend page or return from API.

Reply:
- Can get total by `information` interface
- When an error occurs, error will be returned, instead warning.

Reply:
Get it!